### PR TITLE
Bug 2023571: Remove kube-rbac-proxy for driver metrics

### PR DESCRIPTION
**Pull request by jsafrane**
The driver does not emit any metric.

cc @openshift/storage

### CODE REVIEW

File: assets/controller.yaml [hunk 1]
```diff
@@ -98,28 +98,6 @@ spec:
...
-          - containerPort: 9216
-            name: driver-m
-            protocol: TCP
-          resources:
-            requests:
-              memory: 20Mi
-              cpu: 10m
-          volumeMounts:
-          - mountPath: /etc/tls/private
-            name: metrics-serving-cert
```
Comment (dobsonj) [inline assets/controller.yaml:122]: I think you'll also want to remove the associated volume:
https://github.com/openshift/aws-efs-csi-driver-operator/blob/4310edab61d19bfb9245fed2773d834ff5395b34/assets/controller.yaml#L182-L184

Comment (dobsonj) [reply assets/controller.yaml:122]: And where does the `aws-efs-csi-driver-controller-metrics-serving-cert` secret get created? Does the secret also need to be removed?

Comment (jsafrane) [reply assets/controller.yaml:122]: > I think you'll also want to remove the associated volume:

That volume is still used in `provisioner-kube-rbac-proxy` container. I added it back there.

> And where does the aws-efs-csi-driver-controller-metrics-serving-cert secret get created? 

It's created by OCP as result of annotation `service.beta.openshift.io/serving-cert-secret-name: aws-efs-csi-driver-controller-metrics-serving-cert` in `service.yaml` - _something_ generates TLS cert and creates a Secret for us.

**Pull request merged**